src/util/tumblr_helpers.js

```diff
@@ -7,7 +7,21 @@ import { inject } from './inject.js';
  */
 export const apiFetch = async function (...args) {
   return inject(
-    async (resource, init) => window.tumblr.apiFetch(resource, init),
+    async (resource, init) => {
+      // make sure init is an object we can modify
+      if (typeof init !== 'object' || init === null) {
+        init = {};
+      }
+
+      // make sure init.headers is an object we can modify
+      if (typeof init.headers !== 'object' || init.headers === null) {
+        init.headers = {};
+      }
+
+      // make sure every API request we make is telling Tumblr that we're XKit
+      init.headers['X-XKit'] = '1';
+      window.tumblr.apiFetch(resource, init);
```

Comment:
This needs to be explicitly returned now that it's no longer the sole statement; arrow functions only return implicitly in their braceless form.

```suggestion
      return window.tumblr.apiFetch(resource, init);
```